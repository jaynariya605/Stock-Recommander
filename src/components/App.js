import React, { useReducer } from "react";
import { Header } from "./Header";
import SearchForm from "./SearchForm";
import StockTable from "./StockTable";
import StockHeader from "./StockHeader";
import Application from "../context/Application";
import stockheader from "../reducer/stockheader";

const App = () =>  {
    const [state, dispatch ] = useReducer(stockheader)
    
    return (
        <div>
        <Application.Provider value={ { state, dispatch }} >
            <Header/>
            <StockHeader/>
            <SearchForm/>
            <StockTable/>
        </Application.Provider>
        </div>
    )
}

export default App