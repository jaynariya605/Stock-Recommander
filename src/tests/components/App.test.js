import { shallow } from 'enzyme';
import App from "../../components/App";
import { configure } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

configure({
    adapter: new Adapter()
})

test('should render App correctly', ()=>{
    const wrapper = shallow(<App/>)
    expect(wrapper).toMatchSnapshot()
})