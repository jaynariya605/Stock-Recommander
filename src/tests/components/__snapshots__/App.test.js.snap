// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<div>
  <ContextProvider
    value={
      Object {
        "dispatch": [Function],
        "state": undefined,
      }
    }
  >
    <Header />
    <_default />
    <_default />
    <_default />
  </ContextProvider>
</div>
`;
